GoogleForCreators/web-stories-wp
# Quick Action Menu: Text Element

## Context

Part of new quick action menu feature in editor. 

## Summary

Adds a new menu type for when a single text element is selected on the canvas. Has ability to: 1) navigate to change text color 2) navigate to change font 3) add animation 4) add link 5) if animation is present clear animation

## Relevant Technical Choices

- Renamed the highlight `states.TEXT` to be `states.TEXT_SET` since adding in states for the text style panel makes that bit a little confusing.
- Gave `CHANGE_TEXT_COLOR` it's own copy constant because if copy changes I want to make sure it's an easy update despite right now it being the same as changing shape color. Also makes karma fixture easier to read since we're setting up each button there. 
- 2 new highlights - `TEXT_COLOR` and `FONT` - both go to the text style panel. 
-  Had to forward some refs in the textStyle component 😅   - only added forwardRef to `DropDown` - otherwise just passing it through the levels as a prop since we don't need it until getting to the actual input. `Color` already had a forwardRef I could tap into. 
- Purposefully didn't update the `isPersistable` prop in the StylePanel because we always expand that panel when a text element is either added or selected in the canvas. This lets the highlight work as expected while maintaining existing functionality (shout out Karma tests for being really useful to avoid this regression!)

## To-do

- You'll notice that the font picker doesn't highlight, it does focus though. Once the keyboard update #7623 is merged this will show the focus outline when the keyboard is getting used.
- Something potentially odd is happening in the textStyle tests. It's passing, but if you go to the javascript tests below and just scroll through them a little you will see two warnings like so: `Check the render method of `FontPicker`.` It's got to do something with how this whole panel is getting mocked to test the sub sections of it? I spent some time debugging but the render methods seem fine. If anyone reading this knows what's up please let me know! 

## User-facing changes

Menu is behind a feature flag. 

## Testing Instructions

Make sure the quick actions experiment is active. 
Add a text element to a page, while it is selected you should see the new quick action menu for them. Click the buttons and see that they take you to panels as specified in the summary (or the video below). 


https://user-images.githubusercontent.com/10720454/119906659-7db8c400-bf03-11eb-9a27-bc364c9b814d.mp4



### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6146
